optuna/trial/_frozen.py

```diff
@@ -321,8 +322,15 @@ def _validate(self) -> None:
                     "`datetime_complete` is supposed to be None for an unfinished trial."
                 )
 
+        if self.state == TrialState.PRUNED and self._values is not None:
+            raise ValueError("`value` is supposed not to be set for a pruned trial.")
```

Comment:
```suggestion
            raise ValueError("`value` is not supposed to be set for a pruned trial.")
```
[nit] Fix grammatical error